File: v1_12/src/main/java/me/fullidle/ficore/ficore/v1_12/AbilityWrapper.java
package me.fullidle.ficore.ficore.v1_12;

import com.pixelmonmod.pixelmon.entities.pixelmon.abilities.AbilityBase;

public class AbilityWrapper extends me.fullidle.ficore.ficore.common.api.pokemon.AbilityWrapper<AbilityBase> {
    public AbilityWrapper(AbilityBase original) {
        super(original);
    }

    @Override
    public String getName() {
        return this.getOriginal().getName();
    }

    @Override
    public Class<AbilityBase> getType() {
        return AbilityBase.class;
    }
}

File: v1_21/Cob/src/main/kotlin/me/fullidle/ficore/ficore/vCob_21/AbilityWrapper.kt
package me.fullidle.ficore.ficore.vCob_21

import com.cobblemon.mod.common.api.abilities.AbilityTemplate
import me.fullidle.ficore.ficore.common.api.pokemon.AbilityWrapper

class AbilityWrapper(original: AbilityTemplate) : AbilityWrapper<AbilityTemplate>(original) {
    override fun getName(): String {
        return this.original.name
    }

    override fun getType(): Class<AbilityTemplate> {
        return AbilityTemplate::class.java
    }
}
